types/lib/model.d.ts

```diff
@@ -2911,15 +2911,50 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
   public isSoftDeleted(): boolean;
 }
 
+/**
+ * @deprecated
+ *
+ * Kept for compatibility purposes, but avoid using this.
+ *
+ * - If you're writing a function that consumes a model type, you should likely
+ *   use `ModelStatic<M>` as the input to work well with class and `.define()`
+ *   models.
+ * - If you're writing a function that outputs a model type, you should likely
+ *   output `ModelCtor<M>` to ensure that the model's static properties can
+ *   be called too.
+ */
 export type ModelType<TModelAttributes = any, TCreationAttributes = TModelAttributes> = new () => Model<TModelAttributes, TCreationAttributes>;
 
-// Do not switch the order of `typeof Model` and `{ new(): M }`. For
-// instances created by `sequelize.define` to typecheck well, `typeof Model`
-// must come first for unknown reasons.
+/**
+ * The output of `sequelize.define` and other functions that output a model.
+ *
+ * Do not switch the order of `typeof Model` and `{ new(): M }`. For
+ * instances created by `sequelize.define` to typecheck well, `typeof Model`
+ * must come first for unknown reasons.
```

Comment:
This is probably because `typeof Model` is a constructor and as such, its `new` function overrides the constructor of `{ new(): M }`.